### PR TITLE
Improve shader robust access tests with more memory types.

### PR DESCRIPTION
**Pull request by Kangz**
PTAL!

### CODE REVIEW

**Comment by kainino0x** [comment]
Sorry for the review delay; will get to it Monday

File: src/webgpu/shader/execution/robust_access.spec.ts [hunk 1]
```diff
@@ -295,74 +382,80 @@ g.test('bufferMemory')
...
+      ${globalSource}
+
+      uint runTest() {
+        ${testFunctionSource}
+        return 0;
+      }`;
+
+    // Run it.
+    if (usesBuffer) {
+      // Create a buffer that contains zeroes in the allowed access area, and 42s everywhere else.
```
Comment (kainino0x) [inline src/webgpu/shader/execution/robust_access.spec.ts:434]: Not new, but aren't we supposed to allow reads from anywhere in the buffer to allow Vulkan's robust access behavior?

Comment (Kangz) [reply src/webgpu/shader/execution/robust_access.spec.ts:434]: I'm almost 100% sure that we decided that the array accesses would be clamped.

Comment (kainino0x) [reply src/webgpu/shader/execution/robust_access.spec.ts:434]: ugh, I can't keep anything straight anymore. We've gone through too many proposals on every single little decision in the API. I think you're right.

**Pull request merged**